DogTales/src/player.cpp

```diff
@@ -20,8 +21,8 @@ void Player::handle_wall_collision() {
 	auto const bounce_rect = bave::Rect<>::from_size(m_world_space - m_sprite.get_size(), glm::vec2{0.0f});
 
 	// if the sprite's position exceeds the play area, the corresponding velocity component needs to flip.
-	if (position.x < bounce_rect.top_left().x || position.x > bounce_rect.bottom_right().x) { m_vel.x *= -1.0f; }
-	if (position.y > bounce_rect.top_left().y || position.y < bounce_rect.bottom_right().y) { m_vel.y *= -1.0f; }
+	if (position.x < bounce_rect.top_left().x || position.x > bounce_rect.bottom_right().x) { m_physics.velocity.x *= -0.9f; }
+	if (position.y > bounce_rect.top_left().y || position.y < bounce_rect.bottom_right().y) { m_physics.velocity.y *= -0.9f; }
```

Comment:
Just curious: what's with the 0.9... instead of 1.0? 😄

Reply:
when it's 1.0, it bounces against the ground for a really long time. I think of the 0.9 as the object transferring 10% of its energy into the ground/wall. it's just a needless temporary cosmetic fix, since player collision will probably be completely static anyway :)